[docs] Revise and split up "Overview" page into "Introduction" section (#4692)
mui/mui-x: docs/data/introduction/licensing/licensing.md

# Licensing

<p class="description">MUI X is an open-core, MIT-licensed library. Purchase a commercial license for advanced features and support.</p>

## MIT vs. commercial licenses

MUI has been building MIT-licensed React components since 2014, and we are committed to the continued advancement of the open-source libraries.
You can learn more about our stewardship ethos in [this document from our company handbook](https://mui-org.notion.site/Stewardship-542a2226043d4f4a96dfb429d16cf5bd).

We offer commercial licenses to developers who need the most advanced features that cannot be easily maintained by the open-source community.
Commercial licenses enable us to support a full-time staff of engineers, which is simply not possible through the MIT model.

Rest assured that when we release features commercially, it's only because we believe that you will not find a better MIT-licensed alternative anywhere else.

See the [Pricing](https://mui.com/pricing/) page for a detailed feature comparison.

> ## Community Plan

Reviewer: To keep consistency, since we mention "Pro version" and "Premium version" below. However, https://mui.com/x/react-data-grid/getting-started/#plans uses "plan" for everything.

```suggestion
## Community version
```

Reviewer 2: +1 to use plan for everything.

Author: cc @mbrookes who suggested "version" here. I'm cool with either one, but we should definitely be consistent.

Reviewer 3: > Version could be confused with

Perhaps it's a language barrier thing, but I don't see the confusion. The context makes it clear.

"Plan" feels awkward; "package" feels forced. Perhaps "edition"?

Author: I feel that we should stick with the terminology of Community Plan / Pro Plan / Premium Plan as much as possible.

Reviewer 4: +1 for `Plan` on everything.

Reviewer 2: I have tried to resolve the conflict by using Plan for the headers and version as a synonym to avoid repetition in the body, it seems to work well.

# Licensing

<p class="description">MUI X is an open-core, MIT-licensed library. Purchase a commercial license for advanced features and support.</p>

## MIT vs. commercial licenses

MUI has been building MIT-licensed React components since 2014, and we are committed to the continued advancement of the open-source libraries.
You can learn more about our stewardship ethos in [this document from our company handbook](https://mui-org.notion.site/Stewardship-542a2226043d4f4a96dfb429d16cf5bd).

> We offer commercial licenses to developers who need the most advanced features that cannot be easily maintained by the open-source community.

Reviewer 5: "We offer commercial licenses" 

Isn't it a bit weird to say "offer" for something that we make people pay for ?

Author: No, it makes sense in this context. To "offer" something doesn't imply that it's free. I could offer to sell you my car, to give a random example.

Commercial licenses enable us to support a full-time staff of engineers, which is simply not possible through the MIT model.

> Rest assured that when we release features commercially, it's only because we believe that you will not find a better MIT-licensed alternative anywhere else.

Reviewer 5: Should we add that something MIT will remain MIT forever ?

MIT licensed packages:

- [`@mui/x-data-grid`](https://www.npmjs.com/package/@mui/x-data-grid)
- [`@mui/x-date-pickers`](https://www.npmjs.com/package/@mui/x-date-pickers)

### Pro Plan<span class="plan-pro"></span>

The Pro version of MUI X expands on the features of the free version with more advanced capabilities such as multi-filtering, multi-sorting, column resizing and column pinning for the data grid; as well as the date range picker component.

The Pro version is available under a commercial license—visit [the Pricing page](https://mui.com/pricing/) for details.

Pro packages:

- [`@mui/x-data-grid-pro`](https://www.npmjs.com/package/@mui/x-data-grid-pro)
- [`@mui/x-date-pickers-pro`](https://www.npmjs.com/package/@mui/x-date-pickers-pro)

:::info
The features exclusive to the Pro version are marked with the <span class="plan-pro"></span> icon throughout the documentation.
:::

<div class="only-light-mode">
  <img src="/static/x/commercial-header-icon-light.png" style="width: 579px; margin-bottom: 2rem;" alt="">
</div>
<div class="only-dark-mode">
  <img src="/static/x/commercial-header-icon-dark.png" style="width: 560px; margin-bottom: 2rem;" alt="">
</div>

> ### Premium Plan<span class="plan-premium"></span>

Author: How does that work exactly? Do they pay some kind of prorated fee to cover the difference? cc @joserodolfofreitas

> ### Premium Plan<span class="plan-premium"></span>

Reviewer 4: We'll send a product update campaign next week about it, but I agree it makes totally sense to add it to the docs too.

### How it works

You can use your Pro license as a credit when purchasing MUI X Premium. We’ll provide you with a discount based on the remaining time your current license is valid.
To get your upgrade discount, please contact us at [sales@mui.com](mailto:sales@mui.com?subject=My%20upgrade%20discount%20to%20Premium)

> ### Premium Plan<span class="plan-premium"></span>

Author: Done, thanks!

> The Premium version of MUI X covers the most advanced features of the data grid, such as row grouping, Excel export, aggregation, in addition to everything that's included in the Pro Plan.

Author: ```suggestion
The Premium version of MUI X covers the most advanced features of the data grid, such as row grouping, Excel export, and aggregation, in addition to everything that's included in the Pro Plan.
```

The Premium version is available under a commercial license—visit [the Pricing page](https://mui.com/pricing/) for details.

Premium package:

- [`@mui/x-data-grid-premium`](https://www.npmjs.com/package/@mui/x-data-grid-premium)

:::info

> The features exclusive to the Premium version are marked with the <span class="plan-premium"></span> icon throughout the documentation.

Reviewer 5: Should we add an example image like we do for pro ?

Author: Actually I'm thinking we should remove the screenshot from the Pro section—I don't think it really adds any value.

Reviewer 5: Could also make sense, or just inline an example if it's worth it

:::

## Evaluation (trial) licenses

In accordance with our [End User License Agreement](https://mui.com/store/legal/mui-x-eula/#evaluation-trial-licenses), you can use the Pro and Premium components without a commercial license for 30 days without restrictions.
You don't need to contact us to use these components for evaluation purposes.

You will need to purchase a commercial license in order to remove the watermarks and console warnings, and after the 30-day evaluation period.

## How many developer seats do I need?

The number of seats purchased on your license must correspond to the number of concurrent
developers contributing changes to the front-end code of the projects that uses MUI X Pro or
Premium.

> **Example 1.** Company 'A' is developing an application named 'AppA'. The app needs to

Author: I love these examples!

render 10K rows of data in a table and allow users to group, filter, and sort. The dev team

> adds MUI X Pro to the project to satisfy that requirement. 5 front-end and 10 back-end

Reviewer 3: ```suggestion
adds MUI X Pro to the project to satisfy this requirement. Five front-end and ten back-end
```

> developers are working on 'AppA'. Only 1 developer is tasked with configuring and modifying

Reviewer 3: ```suggestion
developers are working on 'AppA'. Only one developer is tasked with configuring and modifying
```

the data grid. The front-end developers and only are contributing code to the front-end.

> Company 'A' purchases 5 licenses.

Reviewer 3: ```suggestion
Company 'A' purchases five licenses.
```

> **Example 2.** A UI development team at Company 'A' creates its own UI library for

Reviewer 3: ```suggestion
**Example 2.** A UI development team at Company 'B' creates its own UI library for
```

internal development and includes MUI X Pro as a component. The team working on 'AppA' uses
the new library and so does the team working on 'AppB'. 'AppA' has 5 front-end developers

> and 'AppB' has 3. There are 2 front-end developers on the UI development team. Company 'B'

Reviewer 3: ```suggestion
and 'AppB' has three. There are two front-end developers on the UI development team. Company 'B'
```

purchases 10 licenses.

> [The clause in the EULA.](https://mui.com/store/legal/mui-x-eula/#required-quantity-of-licenses)

Reviewer 3: ```suggestion
This is [the relevant clause in the EULA.](https://mui.com/store/legal/mui-x-eula/#required-quantity-of-licenses)
```

The Pro version is available under a commercial license—visit [the Pricing page](https://mui.com/pricing/) for details.

Pro packages:

- [`@mui/x-data-grid-pro`](https://www.npmjs.com/package/@mui/x-data-grid-pro)
- [`@mui/x-date-pickers-pro`](https://www.npmjs.com/package/@mui/x-date-pickers-pro)

:::info
The features exclusive to the Pro version are marked with the <span class="plan-pro"></span> icon throughout the documentation.
:::

### Premium Plan<span class="plan-premium"></span>

The Premium version of MUI X covers the most advanced features of the data grid, such as row grouping, Excel export, and aggregation, in addition to everything that's included in the Pro Plan.

The Premium version is available under a commercial license—visit [the Pricing page](https://mui.com/pricing/) for details.

Premium package:

- [`@mui/x-data-grid-premium`](https://www.npmjs.com/package/@mui/x-data-grid-premium)

:::info
The features exclusive to the Premium version are marked with the <span class="plan-premium"></span> icon throughout the documentation.
:::

## Upgrading from Pro to Premium

You can use your Pro license as a credit when purchasing MUI X Premium.

> We’ll provide you with a discount based on the remaining time that your current license is valid.

Reviewer 2: Straight quotes

```suggestion
We'll provide you with a discount based on the remaining time that your current license is valid.
```

internal development and includes MUI X Pro as a component.
The team working on 'AppA' uses the new library and so does the team working on 'AppB'.
'AppA' has 5 front-end developers and 'AppB' has three.
There are two front-end developers on the UI development team.
Company 'B' purchases ten licenses.

This is [the relevant clause in the EULA.](https://mui.com/store/legal/mui-x-eula/#required-quantity-of-licenses)

## License key installation

When you purchase a commercial license, you'll receive a license key by email.
This key removes all watermarks and console warnings.

:::warning
Licenses purchased after **May 13, 2022** are only compatible with MUI X `v5.11.0` or later.

Please update your package if you're using an earlier version.

If this isn't possible, please contact sales@mui.com to request a compatible license key.
:::

### How to install the key

```jsx
import { LicenseInfo } from '@mui/x-license-pro';

LicenseInfo.setLicenseKey('YOUR_LICENSE_KEY');
```

### Where to install the key

You need to call `setLicenseKey` before React renders the first component.
You only need to install the key once in your application.

:::info

> When using NextJS, you should call `setLicenseKey` in [`_app.js`](https://nextjs.org/docs/advanced-features/custom-app):

Reviewer 2: Official name

```suggestion
When using Next.js, you should call `setLicenseKey` in [`_app.js`](https://nextjs.org/docs/advanced-features/custom-app):
```
